docs/development/coding-style-guide.md

```diff
@@ -0,0 +1,262 @@
+# Lokomotive coding style guide
+
+This document describes the coding style that's enforced in the Lokomotive codebase. The guidelines
+in this document should be followed by Lokomotive maintainers and contributors alike. The guide is
+structured in a way which makes it easy to link to a specific guideline, for example from a PR
+comment.
+
+## Guidelines
+
+This section describes the entire set of coding style guidelines which should be followed when
+working on the Lokomotive codebase.
+
+Each "topic" should reside on its own under a heading so that it's easy to unambiguously link to a
+specific guideline during code reviews.
+
+The list of guidelines is meant to evolve over time. If you think a new guideline should be
+enforced, suggest changes to this document in a PR.
+
+### Error string formatting
```

Comment:
I'd also add about not using `` , rather use `%q` directive.

Reply:
Can add that, although `%q` formats using *double quotes* and sometimes people want single quotes.

Reply:
I think we can add that later, to describe that if you mention a context of error, `%q` should be used or something. Current PR is not opinionated on that as far as I read.

Reply:
Yes, I think @ipochi wanted us to *add* a guideline for `%q`.

Reply:
I'm resolving this for the sake of making progress. @ipochi please feel free to add/modify things in a new PR.